cranelift-codegen/meta/src/gen_legalizer.rs

```diff
@@ -134,6 +134,19 @@ fn unwrap_inst(
                     .get(var_pool.get(def.defined_vars[0]).dst_def.unwrap())
                     .to_comment_string(var_pool)
             ));
+
+            fmt.line("let r = pos.func.dfg.inst_results(inst);");
+            for (i, &var_index) in def.defined_vars.iter().enumerate() {
+                let var = var_pool.get(var_index);
+                fmtln!(fmt, "let {} = &r[{}];", var.name, i);
+                fmtln!(
+                    fmt,
+                    "let typeof_{} = pos.func.dfg.value_type(*{});",
```

Comment:
This is similar to https://github.com/CraneStation/cranelift/pull/833/files#diff-a685da9c9934e511050d2fefc936f786R114

Reply:
Another one for @bnjbvr

Reply:
(answered above)